test/puppetlabs/puppetdb/http/command_test.clj

```diff
@@ -139,3 +136,110 @@
         (let [timestamp (get-in good-msg [:headers :received])]
           (time/parse (time/formatters :date-time) timestamp))))))
 
+(deftest normalize-request
```

Comment:
I think this is actually a handful of different tests. I don't subscribe to the "one assertion per test" rule, but past a couple it gets hard to track down failures.